### ISSUE TITLE
not a valid GroupElement

### ISSUE THREAD

**Issue by lystrosaurus** [stellar/java-stellar-sdk#131]
transaction hash : fe2c8cdd4eb1222e0f3f43b33f1f4fb87f035f31e875a5496701e3c393351ca5
```java
Server server = new Server("https://horizon.stellar.org/");
server.payments()
                  .forTransaction("fe2c8cdd4eb1222e0f3f43b33f1f4fb87f035f31e875a5496701e3c393351ca5").limit(100).execute();
```
```java
java.lang.IllegalArgumentException: not a valid GroupElement
	at net.i2p.crypto.eddsa.math.GroupElement.<init>(GroupElement.java:254)
	at net.i2p.crypto.eddsa.spec.EdDSAPublicKeySpec.<init>(EdDSAPublicKeySpec.java:23)
	at org.stellar.sdk.KeyPair.fromPublicKey(KeyPair.java:114)
	at org.stellar.sdk.KeyPair.fromAccountId(KeyPair.java:105)
	at org.stellar.sdk.responses.KeyPairTypeAdapter.read(KeyPairTypeAdapter.java:19)
	at org.stellar.sdk.responses.KeyPairTypeAdapter.read(KeyPairTypeAdapter.java:11)
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
	at com.google.gson.Gson.fromJson(Gson.java:814)
	at com.google.gson.Gson.fromJson(Gson.java:879)
	at com.google.gson.Gson.fromJson(Gson.java:852)
	at org.stellar.sdk.responses.OperationDeserializer.deserialize(OperationDeserializer.java:26)
	at org.stellar.sdk.responses.OperationDeserializer.deserialize(OperationDeserializer.java:15)
	at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:81)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:60)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
	at com.google.gson.Gson.fromJson(Gson.java:814)
	at com.google.gson.Gson.fromJson(Gson.java:879)
	at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:49)
	at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:19)
	at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
	at com.google.gson.Gson.fromJson(Gson.java:814)
	at com.google.gson.Gson.fromJson(Gson.java:779)
	at com.google.gson.Gson.fromJson(Gson.java:728)
	at org.stellar.sdk.requests.ResponseHandler.handleResponse(ResponseHandler.java:42)
	at org.stellar.sdk.requests.PaymentsRequestBuilder.execute(PaymentsRequestBuilder.java:79)
	at org.stellar.sdk.requests.PaymentsRequestBuilder.execute(PaymentsRequestBuilder.java:116)
```
https://stellarchain.io/address/GAH6H2XPCZS27WMKPTZJPTDN7JMBCDHTLU5WQP7TUI2ORA2M5FY5DHNU
```java
KeyPair pair = KeyPair.fromAccountId("GAH6H2XPCZS27WMKPTZJPTDN7JMBCDHTLU5WQP7TUI2ORA2M5FY5DHNU");
```
```java
java.lang.IllegalArgumentException: not a valid GroupElement

	at net.i2p.crypto.eddsa.math.GroupElement.<init>(GroupElement.java:254)
	at net.i2p.crypto.eddsa.spec.EdDSAPublicKeySpec.<init>(EdDSAPublicKeySpec.java:23)
	at org.stellar.sdk.KeyPair.fromPublicKey(KeyPair.java:114)
	at org.stellar.sdk.KeyPair.fromAccountId(KeyPair.java:105)

```
@bartekn @nullstyle @Synesso @vogel @bjfish @jedmccaleb

**Comment by Synesso** [comment]
This also occurs in Scala SDK, which uses the official `net.i2p.crypto.eddsa` lib v0.3.0

I have raised a bug there under the assumption that we're not doing anything wrong.

https://github.com/str4d/ed25519-java/issues/65

```
scala> stellar.sdk.KeyPair.fromAccountId("GAH6H2XPCZS27WMKPTZJPTDN7JMBCDHTLU5WQP7TUI2ORA2M5FY5DHNU")
java.lang.IllegalArgumentException: not a valid GroupElement
  at net.i2p.crypto.eddsa.math.GroupElement.<init>(GroupElement.java:336)
  at net.i2p.crypto.eddsa.math.GroupElement.<init>(GroupElement.java:287)
  at net.i2p.crypto.eddsa.spec.EdDSAPublicKeySpec.<init>(EdDSAPublicKeySpec.java:36)
  at stellar.sdk.KeyPair$.fromPublicKey(KeyPair.scala:193)
  at stellar.sdk.KeyPair$.fromAccountId(KeyPair.scala:202)
  ... 36 elided
```

**Comment by Synesso** [comment]
Note, I replicated the problem directly on the eddsa library using the decoded byte array for this account id. No fat jars were involved. See https://github.com/str4d/ed25519-java/issues/65

**Comment by Synesso** [comment]
I looked into that a few months ago. The custom jar was created by someone no longer on the team [in this commit, in the previous repo](https://github.com/stellar/java-stellar-base/commit/ce59bfad0954a4a0f565e1889f0d0a7b23612074). I tried to contact the committer for some background, but didn't hear back from him. I decompiled the jar to find the only difference between it and the official one was that the curve table was renamed. After [substituting the official jar and using the new name](https://github.com/Synesso/scala-stellar-sdk/commit/520f0c02f983913c8b4902a50884aa06ebc69223#diff-4cb7ce15e00d093cf0a6b5808c9783ecL136) everything continued to work the same.

**Comment by Synesso** [comment]
Just tried it with the custom jar and it fails in the same way.

**Comment by jillesvangurp** [comment]
Thanks for looking into this. I checked with my shaded jar and can reproduce the problem. So shading /relocating is not the solution for this indeed.

I quickly checked if this works with javascript:

```
var StellarSdk = require('stellar-sdk');

var foo = StellarSdk.Keypair.fromPublicKey("GAH6H2XPCZS27WMKPTZJPTDN7JMBCDHTLU5WQP7TUI2ORA2M5FY5DHNU")
```

I was able to run this with `node index.js` after installing the javascript sdk with npm. So,looks like a bug with the eddsa library. Probably best if you file a bug with them. Also might be worth investigating if there are alternatives to this library for java.

@bartekn I guess the key thing to establish for replacing the jar with the official version is if the unpatched jar is fully compatible with the existing data.
